src/main/java/se/sundsvall/parkingpermit/integration/supportmanagement/mapper/SupportManagementMapper.java

```diff
@@ -71,15 +72,32 @@ public static Errand toSupportManagementCardManagementErrand(generated.se.sundsv
 			.title(SM_SUBJECT_CARD_MANAGEMENT_PARKING_PERMIT)
 			.priority(MEDIUM)
 			.description(String.format(SM_DESCRIPTION_CARD_MANAGEMENT, caseDataErrand.getErrandNumber()))
-			.classification(new Classification().category(SM_CATEGORY_URBAN_DEVELOPMENT).type(SM_TYPE_PARKING_PERMIT))
-			.labels(List.of(SM_LABEL_URBAN_DEVELOPMENT, SM_LABEL_PARKING_PERMIT, SM_LABEL_CARD_MANAGEMENT))
+			.labels(mapLabels(labels, List.of(SM_LABEL_URBAN_DEVELOPMENT, SM_LABEL_PARKING_PERMIT, SM_LABEL_CARD_MANAGEMENT)))
 			.channel(getChannel(isAutomatic))
 			.stakeholders(List.of(getContactStakeholderFromApplicant(caseDataErrand)))
 			.reporterUserId(Optional.ofNullable(administratorStakeholder.getAdAccount()).orElse(PROCESS_ENGINE_USER))
 			.activeNotifications(null)
 			.businessRelated(false);
 	}
 
+	private static List<ErrandLabel> mapLabels(final List<Label> labels, final List<String> labelResources) {
+		if (labels == null || labels.isEmpty()) {
+			return Collections.emptyList();
+		}
+
+		Stream<ErrandLabel> currentLevelStream = labels.stream()
+			.filter(label -> labelResources.contains(label.getResourcePath()))
+			.map(label -> new ErrandLabel().id(label.getId()));
+
+		Stream<ErrandLabel> subLabelsStream = labels.stream()
+			.map(Label::getLabels)
+			.map(subLabels -> mapLabels(subLabels, labelResources))
+			.flatMap(List::stream);
+
+		return Stream.concat(currentLevelStream, subLabelsStream)
+			.collect(Collectors.toList());
```

Comment:
An alternative is to create a recursive method for this to be independent of how the label tree is structured (if the structure dept is changed for example).

Reply:
This method is recursive. What do you mean?